apps/dashboard/src/components/conditions-editor/conditions-editor-context.tsx

```diff
@@ -60,14 +57,14 @@ export function ConditionsEditorProvider({
         return null;
       };
 
-      return findParent(query);
+      return findParent(queryRef.current);
     },
-    [query]
+    [queryRef]
   );
 
   const contextValue = useMemo(
-    () => ({ query, setQuery, removeRuleOrGroup, cloneRuleOrGroup, getParentGroup }),
-    [query, setQuery, removeRuleOrGroup, cloneRuleOrGroup, getParentGroup]
+    () => ({ removeRuleOrGroup, cloneRuleOrGroup, getParentGroup }),
```

Comment:
These changes are part of improving the performance of the conditions editor context. 
The `react-query builder` and `react-hook-form` don't integrate well together. I noticed the performance issues with fields lagging when the form has many rows and rules, but I was able to identify issues and improve it.